### PR TITLE
[Issue #143] Doc tweaks and improvements

### LINKED ISSUE #143: Bit of doc restructuring
Just some notes for tweaking the docs:

* [x] Maintenance section for developers, with things like when / how to update the swagger spec
* [x] Use cases to replace interesting uses, definitely more for jupyter-websocket mode, point directly to demos, point to the messaging spec
* [x] Admonition about the notebook-http mode maybe moving to a plugin / separate repo one day
* [X] ~~Intended audience is developers, say it~~
* [X] ~~Move features to the front page maybe?~~
* [x] Add docker ports, failure to start with seed notebook to troubleshooting
* [x] Separate HTTP provisioning API from Websocket messaging API on jupyter-websocket page
* [x] Everyone loves README badges!
* [x] More install / run recipes

### PR DESCRIPTION
**Pull request by parente**
* Fixes #143
* make docs target
* Cleanup docs/environment.yml
* More badges

### CODE REVIEW

**Comment by parente** [comment]
@willingc Would appreciate your review when you've got time.

**Comment by parente** [comment]
Rebased on master.

**Comment by willingc** [comment]
@parente Reviewing now. Thanks for rebasing.

File: Makefile [hunk 1]
```diff
@@ -29,9 +29,12 @@ clean: ## Remove all built files from the host (but not the dev docker image)
 dev: ARGS?=
 dev: PYARGS?=
 dev: ## Start kernel gateway on port 8888 in a docker container
-	@$(DOCKER) -p 8888:8888 $(IMAGE) \
+	$(DOCKER) -p 8888:8888 $(IMAGE) \
 		python $(PYARGS) kernel_gateway --KernelGatewayApp.ip='0.0.0.0' $(ARGS)
 
+docs: ## Build the sphinx documentation in a container
```
Comment (willingc) [inline Makefile]: Nice!

File: docs/source/devinstall.md [hunk 2]
```diff
@@ -1,25 +1,19 @@
-## Development Installation
+## Development Workflow
 
-This document gives instructions for setup of a Dockerized development 
-environment for the Jupyter Kernel Gateway.
+This document includes instructions for setting up Dockerized development environment for the Jupyter Kernel Gateway. It also includes common steps in the developer workflow such as running tests, building doc, updating specs, etc.
```
Comment (willingc) [inline docs/source/devinstall.md]: ...building docs

**Comment by willingc** [comment]
@parente This looks amazing. One small typo but I'm merging this as is.

**Pull request merged**